File: idea-plugin/src/main/java/com/zzf/rikki/session/SessionStatus.java
package com.zzf.rikki.session;

import java.util.concurrent.ConcurrentHashMap;

public class SessionStatus {
    public static final class Info {
        public final String type;
        public final Integer attempt;
        public final String message;
        public final Long next;

        public Info(String type) {
            this(type, null, null, null);
        }

        public Info(String type, Integer attempt, String message, Long next) {
            this.type = type;
            this.attempt = attempt;
            this.message = message;
            this.next = next;
        }
    }

    private final ConcurrentHashMap<String, Info> state = new ConcurrentHashMap<>();

    public Info get(String sessionId) {
        return state.getOrDefault(sessionId, new Info("idle"));
    }

    public void set(String sessionId, Info info) {
        if (info == null || "idle".equals(info.type)) {
            state.remove(sessionId);
        } else {
            state.put(sessionId, info);
        }
    }
}

File: idea-plugin/src/main/kotlin/com/zzf/rikki/idea/agent/compat/SessionStatus.kt
package com.zzf.rikki.idea.agent.compat

import java.util.concurrent.ConcurrentHashMap

data class SessionStatusInfo(
    val type: String,
    val attempt: Int? = null,
    val message: String? = null,
    val next: Long? = null
)

class SessionStatus {
    private val state = ConcurrentHashMap<String, SessionStatusInfo>()

    fun get(sessionId: String): SessionStatusInfo = state[sessionId] ?: SessionStatusInfo(type = "idle")

    fun set(sessionId: String, info: SessionStatusInfo) {
        if (info.type == "idle") {
            state.remove(sessionId)
        } else {
            state[sessionId] = info
        }
    }
}